### PR TITLE
LPAL-256 Improve contrast

### PR DESCRIPTION
**Pull request by townxelliot**
## Purpose

https://opgtransform.atlassian.net/browse/LPAL-256

Fixes LPAL-256

## Approach

* Upgraded cypress-axe to improve parity between browser extension results and axe-core test automation results.
* Patch govuk SASS files which need to be overridden in our app during the grunt build.
* Remove more files checked into the code base which are interim files generated by grunt.
* Add overrides for colours where required. As we are using old govuk templates, the colours provide insufficient contrast, so I modified our SASS to perform overrides, mostly using !important to ensure they take precedence. I also did global replacements for focus and button colours across the generated CSS.

## Learning

- Learned more about how the SASS and CSS files are organised for the app.
- Learned how to use the Axe and Wave Chrome extensions to check page accessibility.
- Learned more about axe-core's API to enable me to selectively run tests for specific issues and provide better logging when accessibility violations occur.

## Checklist

* [X] I have performed a self-review of my own code
* [X] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [X] I have added tests to prove my work
* [x] The product team have tested these changes

### CODE REVIEW

File: cypress/support/commands.js [hunk 1]
```diff
@@ -24,26 +24,58 @@
...
+
+        location = `${location}`.replace(Cypress.config('baseUrl'), '');
+        cy.task('log', `\n************** ACCESSIBILITY VIOLATIONS ON ${location}`);
+
+        reports.forEach((report) => {
+            cy.task('log', `-------------- ID: ${report.id}`);
+            cy.task('log', `Impact: ${report.impact}`);
+            cy.task('log', `Description: ${report.description}`);
+            cy.task('log', 'HTML elements causing violation:')
+            cy.task('log', '* ' + report.snippets.join('\n*  '));
```
Comment (townxelliot) [inline cypress/support/commands.js:75]: normalise spaces

File: service-front/Gruntfile.js [hunk 2]
```diff
@@ -60,15 +60,18 @@ module.exports = function (grunt) {
       }
     },
 
-    // govuk template css copied into public/assets
+    // govuk template css - copy our fork of the template to public dir, as we
```
Comment (townxelliot) [inline service-front/Gruntfile.js:63]: comment that we will not need this later when we are using the CSS directly from the node module

File: service-front/Gruntfile.js [hunk 3]
```diff
@@ -60,15 +60,18 @@ module.exports = function (grunt) {
...
+    // overwrite our custom changes...
     copy: {
       main: {
         files: [{
           expand: true,
           src: [
-            'node_modules/govuk_template_mustache/assets/stylesheets/fonts.css',
-            'node_modules/govuk_template_mustache/assets/stylesheets/govuk-template-print.css',
-            'node_modules/govuk_template_mustache/assets/stylesheets/govuk-template.css'
+            'assets/stylesheets/fonts.css',
```
Comment (townxelliot) [inline service-front/Gruntfile.js:72]: can we do this replacement on the fly using grunt?

File: service-front/assets/sass/settings/_variables.scss [hunk 4]
```diff
@@ -15,3 +15,9 @@ $status-returned-colour: #28a197;
 
 $progress-bar-background: #dadada;
 $progress-bar-font-default: #555;
+
+$focus-colour-override: #ffdd00;
```
Comment (townxelliot) [inline service-front/assets/sass/settings/_variables.scss:19]: comment that these match the latest design system colours

File: service-front/assets/sass/patterns/_banner.scss [hunk 5]
```diff
@@ -40,8 +40,8 @@
    ========================================================================== */
 
 .moj-banner--success {
-  border-color: #00823b;
-  color: #00823b;
+  border-color: #00703c;
```
Comment (townxelliot) [inline service-front/assets/sass/patterns/_banner.scss:43]: should reference variable value $button-colour-override?

**Review by nickdavis2001 — DISMISSED** [review]
lgtm

**Review by nickdavis2001 — APPROVED** [review]

**Pull request merged**